walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,79 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
+  int FULL_DAY = 24*60;
+  Boolean[] freeStartTimes = new Boolean[FULL_DAY];
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    initializeFreeStartTimes();
+
+    for(Event e: events) {
+      if(!Collections.disjoint(e.getAttendees(), request.getAttendees())) {
+        busy(e, (int)request.getDuration());
+      }
+    }
+
+    return createFreeTimeRanges((int)request.getDuration());
+  }
+
+  private void initializeFreeStartTimes() {
+    Arrays.fill(freeStartTimes, Boolean.TRUE);
+  }
+
+  private void busy(Event e, int duration) {
+    for(int i = (e.getWhen().start()-duration); i < e.getWhen().end(); ++i) {
+      if(i < 0) {
+        i = 0;
+      }
+      if(i==(8*60+30)) {
```

Comment:
Define this magic number (510) as a const.

Reply:
This was an oversight by me, I removed this but forgot to push the last commit. I used this to debug but it does not have any practical use